packages/core/src/model.ts

```diff
@@ -736,7 +765,7 @@ export class Model implements Storage {
     if (!this.isMixedIn(doc, mixin)) {
       throw new Error(`Class ${doc._class} could not be cast to ${mixin}`)
     }
-    const proxy = Object.create(this.getPrototype(mixin)) as Proxy & T
+    const proxy = Object.create(this.getPrototype(mixin), this.getProperties(mixin)) as Proxy & T
```

Comment:
Please reuse properties from getPrototype() call, I think we do not need double them. Thanks.